src/snowflake/cli/plugins/nativeapp/codegen/snowpark/extension_function_utils.py

```diff
@@ -0,0 +1,368 @@
+    ex_fn["anonymous"] = ex_fn.get("anonymous", False)
+    ex_fn["replace"] = ex_fn.get("replace", False)
+    ex_fn["if_not_exists"] = ex_fn.get("if_not_exists", False)
+
+    if ex_fn["replace"] and ex_fn["if_not_exists"]:
+        raise MalformedExtensionFunctionError(
+            "Options 'replace' and 'if_not_exists' are incompatible."
+        )
+
+    default_input_args: List[Dict[str, Any]] = []
+    _sanitize_list_or_dict_attribute(
+        ex_fn=ex_fn,
+        attr="input_args",
+        expected_type=list,
+        default_value=default_input_args,
+    )
+    default_input_types: List[str] = []
+    _sanitize_list_or_dict_attribute(
+        ex_fn=ex_fn,
+        attr="input_sql_types",
+        expected_type=list,
+        default_value=default_input_types,
+    )
+    if len(ex_fn["input_args"]) != len(ex_fn["input_sql_types"]):
+        raise MalformedExtensionFunctionError(
+            "The number of extension function parameters does not match the number of parameter types."
+        )
+
+    _sanitize_str_attribute(ex_fn=ex_fn, attr="all_imports")
+    _sanitize_str_attribute(ex_fn=ex_fn, attr="all_packages")
+    _sanitize_list_or_dict_attribute(
+        ex_fn=ex_fn, attr="external_access_integrations", expected_type=list
+    )
+    _sanitize_list_or_dict_attribute(ex_fn=ex_fn, attr="secrets", expected_type=dict)
+    _sanitize_str_attribute(ex_fn=ex_fn, attr="inline_python_code")
+    _sanitize_str_attribute(ex_fn=ex_fn, attr="execute_as", make_uppercase=True)
+    _sanitize_str_attribute(ex_fn=ex_fn, attr="handler")
+    _sanitize_str_attribute(
+        ex_fn=ex_fn, attr="runtime_version", py_file=py_file, raise_err=True
+    )
+
+    has_app_roles = (
+        ex_fn.get("application_roles", None) and len(ex_fn["application_roles"]) > 0
+    )
+    if has_app_roles:
+        if all(isinstance(app_role, str) for app_role in ex_fn["application_roles"]):
+            ex_fn["application_roles"] = [
+                app_role.upper() for app_role in ex_fn["application_roles"]
+            ]
+        else:
+            raise MalformedExtensionFunctionError(
+                f"Attribute 'application_roles' of extension function must be a list of strings."
+            )
+    else:
+        ex_fn["application_roles"] = []
+
+
+def _get_handler_path_without_suffix(
+    file_path: Path, deploy_root: Path, suffix_str_to_rm: Optional[str] = None
+) -> str:
+    """
+    Get a handler for an extension function based on the file path on the stage. If a specific suffix needs to be removed from the path,
+    then that is also taken into account.
+    """
+    # rel_file_path = file_path.absolute().relative_to(deploy_root.resolve())  # No leading slash
+    # stem, suffix = os.path.splitext(rel_file_path)
+    # if (suffix_str_to_rm is not None) and (suffix == suffix_str_to_rm):
+    #     file_parts = Path(stem).parts
+    # else:
+    #     file_parts = rel_file_path.parts
+    # return ".".join(file_parts)
```

Comment:
What's happening with these commented-out lines?

Reply:
The handler work is a bit more complicated, and is being resolved as part of the follow up PR. Until then, it stays "NotImplementedHandler". I have removed the commented lines to avoid confusion, but now you have function arguments that are essentially unused, but I can leave them as such for now.